plugins/module_utils/network/sonic/config/bfd/bfd.py

```diff
@@ -0,0 +1,790 @@
+
+        if profiles:
+            url = '%s/openconfig-bfd-ext:bfd-profile/profile' % (BFD_PATH)
+            requests.append({'path': url, 'method': DELETE})
+        if single_hops:
+            url = '%s/openconfig-bfd-ext:bfd-shop-sessions/single-hop' % (BFD_PATH)
+            requests.append({'path': url, 'method': DELETE})
+        if multi_hops:
+            url = '%s/openconfig-bfd-ext:bfd-mhop-sessions/multi-hop' % (BFD_PATH)
+            requests.append({'path': url, 'method': DELETE})
+
+        return requests
+
+    def get_delete_profile_request(self, profile_name):
+        url = '%s/openconfig-bfd-ext:bfd-profile/profile=%s' % (BFD_PATH, profile_name)
+        request = {'path': url, 'method': DELETE}
+
+        return request
+
+    def get_delete_profile_attr_request(self, profile_name, attr):
+        url = '%s/openconfig-bfd-ext:bfd-profile/profile=%s/config/%s' % (BFD_PATH, profile_name, attr)
+        request = {'path': url, 'method': DELETE}
+
+        return request
+
+    def get_delete_shop_request(self, remote_address, interface, vrf, local_address):
+        url = '%s/openconfig-bfd-ext:bfd-shop-sessions/single-hop=%s,%s,%s,%s' % (BFD_PATH, remote_address, interface, vrf, local_address)
+        request = {'path': url, 'method': DELETE}
+
+        return request
+
+    def get_delete_shop_attr_request(self, remote_address, interface, vrf, local_address, attr):
+        url = '%s/openconfig-bfd-ext:bfd-shop-sessions/single-hop=%s,%s,%s,%s/config/%s' % (BFD_PATH, remote_address, interface, vrf, local_address, attr)
+        request = {'path': url, 'method': DELETE}
+
+        return request
+
+    def get_delete_mhop_request(self, remote_address, vrf, local_address):
+        url = '%s/openconfig-bfd-ext:bfd-mhop-sessions/multi-hop=%s,null,%s,%s' % (BFD_PATH, remote_address, vrf, local_address)
+        request = {'path': url, 'method': DELETE}
+
+        return request
+
+    def get_delete_mhop_attr_request(self, remote_address, vrf, local_address, attr):
+        url = '%s/openconfig-bfd-ext:bfd-mhop-sessions/multi-hop=%s,null,%s,%s/config/%s' % (BFD_PATH, remote_address, vrf, local_address, attr)
+        request = {'path': url, 'method': DELETE}
+
+        return request
+
+    def get_profile_name(self, profile_name):
+        return profile_name.get('profile_name')
+
+    def sort_lists_in_config(self, config):
+        if 'profiles' in config and config['profiles'] is not None:
+            config['profiles'].sort(key=self.get_profile_name)
+        if 'single_hops' in config and config['single_hops'] is not None:
+            config['single_hops'].sort(key=lambda x: (x['remote_address'], x['interface'], x['vrf'], x['local_address']))
+        if 'multi_hops' in config and config['multi_hops'] is not None:
+            config['multi_hops'].sort(key=lambda x: (x['remote_address'], x['vrf'], x['local_address']))
+
+    def remove_default_entries(self, data):
+        new_data = {}
```

Comment:
Is it possible to use dict pop() function to avoid creating new_data?

Reply:
Yes, I've updated the code.

Reply:
If there is the reason behind of the design, it is not necessary to chang. Anyway either way works.